Inventory.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System.Text;
+
+public class Inventory
+{
+    private List<Product> _products = new();
+
+    public void AddProduct(Product newProduct)
+    {
+        _products.Add(newProduct);
+    }
+
+    public bool IsEmpty()
+    {
+        return _products.Count == 0;
+    }
+
+    public string PrintAllProducts()
+    {
+        if (IsEmpty())
+            return "There are no products.";
+
+        var allProducts = new StringBuilder();
+
+        for (var i = 0; i < _products.Count; i++)
+        {
+            allProducts.Append($"""
+                Product #{i + 1}:
+                {_products[i].ToString()}
+                
+
+                """);
+        }
+
+        return allProducts.ToString();
+    }
+
+    public void EditProductName(Product product, string? newName)
+    {
+        product.Name = newName;
+    }
+
+    public void EditProductPrice(Product product, decimal newPrice)
+    {
+        product.Price = newPrice;
+    }
+
+    public void EditProductQuantity(Product product, int newQuantity)
```

Comment:
Insted of having more than one function for editing the product you can insted do the following  : 

1. Make an `EditProduct` function that takes a product , price , name  and quantity .
2. In the main you first search the product using `SearchProduct` function .
3. Then if it found a product you can pass it to the `EditProduct` function .